frontend/src/app/components/block/block.component.ts

```diff
@@ -170,13 +170,9 @@ export class BlockComponent implements OnInit, OnDestroy {
         this.transactions = null;
         this.transactionsError = null;
         this.isLoadingOverview = true;
-        this.isAwaitingOverview = true;
-        this.overviewError = true;
-        if (this.blockGraph) {
-          this.blockGraph.exit(direction);
-        }
+        this.overviewError = null;
       }),
-      debounceTime(300),
+      throttleTime(300, asyncScheduler, { leading: true, trailing: true }),
```

Comment:
I'm inclined to keep the debounce behavior, because otherwise holding down the the left/right arrow keys to quickly scroll through blocks generates a huge number of requests and lots of busy-work.

We can avoid the initial 300ms delay by switching to `throttleTime` with this config, which resolves the first change immediately, then debounces subsequent changes.

Reply:
This works great, it's the best of both worlds :)